File: Patterns/Facade Pattern/Cpp/HomeTheater/TheaterLights.cpp
#pragma once

#include <iostream>
#include <string>

using namespace std;

class TheaterLights {
private:
    string description;

public:
    TheaterLights(string description) : description{description} {}
    
    void on() {
        cout << description << " on\n";
    }

    void off() {
        cout << description << " off\n";
    }

    void dim(int level) {
        cout << description << " dimming to " << level << "\n";
    }

    string toString() {
        return description;
    }
};

File: Patterns/Facade Pattern/CSharp/HomeTheater/TheaterLights.cs
using System;

namespace HomeTheater
{
    public class TheaterLights
    {
        private string description;

        public TheaterLights(string description)
        {
            this.description = description;
        }

        public void On() => Console.WriteLine($"{description} on");

        public void Off() => Console.WriteLine($"{description} off");

        public void Dim(int level) => Console.WriteLine($"{description} dimming to {level}");

        public string Description { get => description; private set => description = value; }
    }
}